sql/updates/latest-dev.sql

```diff
@@ -442,3 +442,79 @@ BEGIN
+$BODY$
+DECLARE
+    chunk_row _timescaledb_catalog.chunk;
+    hypertable_row _timescaledb_catalog.hypertable;
+    constraint_oid OID;
+    constraint_type CHAR;
+    check_sql TEXT;
+    def TEXT;
+    indx_tablespace NAME;
+    tablespace_def TEXT;
+BEGIN
+    SELECT * INTO STRICT chunk_row FROM _timescaledb_catalog.chunk c WHERE c.id = chunk_constraint_row.chunk_id;
+    SELECT * INTO STRICT hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.id = chunk_row.hypertable_id;
+
+    IF chunk_constraint_row.dimension_slice_id IS NOT NULL THEN
+	    RAISE 'cannot create dimension constraint %', chunk_constraint_row;
+    ELSIF chunk_constraint_row.hypertable_constraint_name IS NOT NULL THEN
+
+        SELECT oid, contype INTO STRICT constraint_oid, constraint_type FROM pg_constraint
+        WHERE conname=chunk_constraint_row.hypertable_constraint_name AND
+              conrelid = format('%I.%I', hypertable_row.schema_name, hypertable_row.table_name)::regclass::oid;
+
+        IF constraint_type IN ('p','u') THEN
+          -- since primary keys and unique constraints are backed by an index
+          -- they might have an index tablespace assigned
+          -- the tablspace is not part of the constraint definition so
+          -- we have to append it explicitly to preserve it
+          SELECT T.spcname INTO indx_tablespace
+          FROM pg_constraint C, pg_class I, pg_tablespace T
+          WHERE C.oid = constraint_oid AND C.contype IN ('p', 'u') AND I.oid = C.conindid AND I.reltablespace = T.oid;
+
+          def := pg_get_constraintdef(constraint_oid);
+
+        ELSIF constraint_type = 't' THEN
+          -- constraint triggers are copied separately with normal triggers
+          def := NULL;
+        ELSE
+          def := pg_get_constraintdef(constraint_oid);
+        END IF;
+
+    ELSE
+        RAISE 'unknown constraint type';
+    END IF;
+
+    IF def IS NOT NULL THEN
+        -- to allow for custom types with operators outside of pg_catalog
+        -- we set search_path to @extschema@
+        SET LOCAL search_path TO @extschema@, pg_temp;
+        EXECUTE pg_catalog.format(
+            $$ ALTER TABLE %I.%I ADD CONSTRAINT %I %s $$,
+            chunk_row.schema_name, chunk_row.table_name, chunk_constraint_row.constraint_name, def
+        );
+
+        -- if constraint (primary or unique) needs a tablespace then add it
+        -- via a separate ALTER INDEX SET TABLESPACE command. We cannot append it
+        -- to the "def" string above since it leads to a SYNTAX error when
+        -- "DEFERRABLE" or "INITIALLY DEFERRED" are used in the constraint
+        IF indx_tablespace IS NOT NULL THEN
+            EXECUTE pg_catalog.format(
+                $$ ALTER INDEX %I.%I SET TABLESPACE %I $$,
+                chunk_row.schema_name, chunk_constraint_row.constraint_name, indx_tablespace
+            );
+        END IF;
+
+    END IF;
+END
+$BODY$ SET search_path TO pg_catalog, pg_temp;
```

Comment:
Do we need it in latest-dev.sql?? You're not changing the function signature neither the return type, so don't need it because it will be replaced by CREATE OR REPLACE on chunk_constraint.sql

Reply:
@fabriziomello depends on when `chunk_constraint.sql` gets called in the update process. If it's called BEFORE then we do need this.

Reply:
@fabriziomello removed it from latest-dev.sql